taichi/transforms/compile_to_offloads.cpp

```diff
@@ -48,6 +48,9 @@ void compile_to_offloads(IRNode *ir,
   print("Typechecked");
   irpass::analysis::verify(ir);
 
+  irpass::demote_operations(ir);
+  print("Operations Demoted");
+
```

Comment:
Let's move this to just before `Simplified IV` as it can impede other optimizations.

Reply:
So, we only demote_operations in `compile_to_executable`, not `compile_to_offloads`?

Reply:
Yes. We may perform some optimizations after `compile_to_offloads` in async mode.